### ISSUE TITLE
In development when using ESM version messages are not "compiled"

### ISSUE THREAD

**Issue by MartijnHols** [lingui/js-lingui#1063]
**Describe the bug**
In development, before extracting new messages, translations are not being compiled meaning variables are not being replaced with their values.

**To Reproduce**
Set up a project that uses the ESM imports in development mode.

```
import { Trans } from "@lingui/react"

export default function App() {
   const somethingElse = 'replaced'
   return <Trans id="my.id">This should be {somethingElse}!</Trans>
}
```

This will render `This should be {somethingElse}!` when using the development ESM version of the library @lingui/core (`dev.esm.js`). This becomes `This should be replaced!` If you switch to the CJS version (`core.development.js` or `dev.development.js`).

**Expected behavior**
The ESM versions works the same as the CJS version.

**Additional context**
The issue seems to be that these lines:
https://github.com/lingui/js-lingui/blob/ff7a235b4cb5ae016650c008a9ce7ac3b3defb0c/packages/core/src/i18n.ts#L188-L192
Are missing from the generated ESM file:
```js
     {
        key: '_',
        value: function (e) {
          var t =
              arguments.length > 1 && void 0 !== arguments[1]
                ? arguments[1]
                : {},
            r =
              arguments.length > 2 && void 0 !== arguments[2]
                ? arguments[2]
                : {},
            n = r.message,
            a = r.formats
          i(e) || ((t = e.values || t), (n = e.message), (e = e.id))
          var o = this.messages[e] || n || e,
            l = this._missing
          return l && !this.messages[e]
            ? s(l)
              ? l(this.locale, e)
              : l
            : (this.messages[e] ||
                this.emit('missing', { id: e, locale: this._locale }),
              i(o) && /\\u[a-fA-F0-9]{4}/g.test(o)
                ? JSON.parse('"'.concat(o, '"'))
                : i(o)
                ? o
                : d(o, this.locale, this.locales, this.localeData)(t, a))
        },
      },
```

I pressume this is because `dev.esm.js` is generated using a `production` value for the NODE_ENV, rather than development. Looking at https://github.com/lingui/js-lingui/blob/ff7a235b4cb5ae016650c008a9ce7ac3b3defb0c/scripts/build/rollup.js#L99 there doesn't seem to be a development ESM build.

- jsLingui version 3.8

ps. the fact that the ESM files are uglified makes debugging harder, and makes the components have unreadable component names in the React Developer Tools and stack traces.

**Comment by semoal** [comment]
https://github.com/lingui/js-lingui/pull/1059
It's already fixed, on the next tuesday will be released

**Comment by semoal** [comment]
Released 3.9.0 give it a try when you can, but i'm pretty sure that will fixed :)
